subsys/net/conn_mgr/conn_mgr.c

```diff
@@ -31,118 +31,126 @@ K_SEM_DEFINE(conn_mgr_event_signal, 1, 1);
-{
-	if (iface_states[index] & CONN_MGR_IF_UP) {
-		return NET_CONN_MGR_STATE_CONNECTED;
-	}
-
-	return NET_CONN_MGR_STATE_DISCONNECTED;
-}
-
 #if defined(CONFIG_NET_IPV6)
-static enum net_conn_mgr_state conn_mgr_ipv6_status(int index)
+static bool conn_mgr_is_if_ipv6_ready(int index)
 {
-	if ((iface_states[index] & CONN_MGR_IPV6_STATUS_MASK) ==
-	    CONN_MGR_IPV6_STATUS_MASK) {
+	if ((iface_states[index] & CONN_MGR_IPV6_STATUS_MASK) == CONN_MGR_IPV6_STATUS_MASK) {
 		NET_DBG("IPv6 connected on iface index %u", index + 1);
-		return NET_CONN_MGR_STATE_CONNECTED;
+		return true;
 	}
 
-	return NET_CONN_MGR_STATE_DISCONNECTED;
+	return false;
 }
 #else
-#define conn_mgr_ipv6_status(...) NET_CONN_MGR_STATE_CONNECTED
+#define conn_mgr_is_if_ipv6_ready(...) false
 #endif /* CONFIG_NET_IPV6 */
 
 #if defined(CONFIG_NET_IPV4)
-static enum net_conn_mgr_state conn_mgr_ipv4_status(int index)
+static bool conn_mgr_is_if_ipv4_ready(int index)
 {
-	if ((iface_states[index] & CONN_MGR_IPV4_STATUS_MASK) ==
-	    CONN_MGR_IPV4_STATUS_MASK) {
+	if ((iface_states[index] & CONN_MGR_IPV4_STATUS_MASK) == CONN_MGR_IPV4_STATUS_MASK) {
 		NET_DBG("IPv4 connected on iface index %u", index + 1);
-		return NET_CONN_MGR_STATE_CONNECTED;
+		return true;
 	}
 
-	return NET_CONN_MGR_STATE_DISCONNECTED;
+	return false;
 }
 #else
-#define conn_mgr_ipv4_status(...) NET_CONN_MGR_STATE_CONNECTED
+#define conn_mgr_is_if_ipv4_ready(...) false
 #endif /* CONFIG_NET_IPV4 */
 
-static void conn_mgr_notify_status(int index)
+/**
+ * @brief Retrieves pointer to an iface by the index that corresponds to it in iface_states
+ *
+ * @param index - The index in iface_states to find the corresponding iface for.
+ * @return net_if* - The corresponding iface.
+ */
+static struct net_if *conn_mgr_get_if_by_index(int index)
 {
-	struct net_if *iface = net_if_get_by_index(index + 1);
+	return net_if_get_by_index(index + 1);
+}
+
+/**
+ * @brief Notifies listeners of the current readiness state of the iface at the given index
+ *
+ * @param index - Index of the iface (in iface_states)
+ */
+static void conn_mgr_notify_if_readiness(int index)
+{
+	struct net_if *iface = conn_mgr_get_if_by_index(index);
+	bool readiness = iface_states[index] & CONN_MGR_IF_READY;
 
 	if (iface == NULL) {
 		return;
 	}
 
-	if (iface_states[index] & CONN_MGR_IF_READY) {
-		NET_DBG("Iface %d (%p) connected",
-			net_if_get_by_iface(iface), iface);
-		net_mgmt_event_notify(NET_EVENT_L4_CONNECTED, iface);
-	} else {
-		NET_DBG("Iface %d (%p) disconnected",
-			net_if_get_by_iface(iface), iface);
-		net_mgmt_event_notify(NET_EVENT_L4_DISCONNECTED, iface);
+	NET_DBG("Iface %d (%p) %s", net_if_get_by_iface(iface),
+		iface, readiness ? "ready" : "unready");
+
+	net_mgmt_event_notify(
+		readiness ? NET_EVENT_L4_CONNECTED : NET_EVENT_L4_DISCONNECTED,
+		iface
+	);
+
+}
+
+static void conn_mgr_set_ready(int idx, bool readiness)
+{
+	/* Clear and then update the L4-readiness bit */
+	iface_states[idx] &= ~CMGR_IF_READY;
```

Comment:
nit: This commit could already use proper flags, instead of having them fixed in the next one.

Reply:
I'm not 100% sure which commit you commented on here, so just to clarify, you are talking about https://github.com/zephyrproject-rtos/zephyr/pull/56549/commits/02c02bb5de16a004c18b2c62ad2d8683faffc7a4 right?

Reply:
Whoops. I've had some troubles with propagating my event renames. Not sure how those changes keep ending up in later commits 🤦. Thanks for catching that, I'll get it corrected.